crates/router/src/connector/rapyd.rs

```diff
@@ -0,0 +1,664 @@
+            message: response.status.status,
+            reason: response.status.message,
+        })
+    }
+}
+
+impl api::PaymentSync for Rapyd {}
+impl
+    services::ConnectorIntegration<api::PSync, types::PaymentsSyncData, types::PaymentsResponseData>
+    for Rapyd
+{
+    fn get_headers(
+        &self,
+        _req: &types::PaymentsSyncRouterData,
+        _connectors: &settings::Connectors,
+    ) -> CustomResult<Vec<(String, String)>, errors::ConnectorError> {
+        Ok(vec![])
+    }
+
+    fn get_content_type(&self) -> &'static str {
+        "application/json"
+    }
+
+    fn get_url(
+        &self,
+        _req: &types::PaymentsSyncRouterData,
+        _connectors: &settings::Connectors,
+    ) -> CustomResult<String, errors::ConnectorError> {
+        Err(errors::ConnectorError::NotImplemented("PSync".to_string()).into())
+    }
+
+    fn build_request(
+        &self,
+        _req: &types::PaymentsSyncRouterData,
+        _connectors: &settings::Connectors,
+    ) -> CustomResult<Option<services::Request>, errors::ConnectorError> {
+        Ok(None)
+    }
+
+    fn get_error_response(
+        &self,
+        _res: Bytes,
+    ) -> CustomResult<ErrorResponse, errors::ConnectorError> {
+        Err(errors::ConnectorError::NotImplemented("PSync".to_string()).into())
+    }
+
+    fn handle_response(
+        &self,
+        _data: &types::PaymentsSyncRouterData,
+        _res: Response,
+    ) -> CustomResult<types::PaymentsSyncRouterData, errors::ConnectorError> {
+        Err(errors::ConnectorError::NotImplemented("PSync".to_string()).into())
+    }
+}
+
+impl api::PaymentCapture for Rapyd {}
+impl
+    services::ConnectorIntegration<
+        api::Capture,
+        types::PaymentsCaptureData,
+        types::PaymentsResponseData,
+    > for Rapyd
+{
+    fn get_headers(
+        &self,
+        _req: &types::PaymentsCaptureRouterData,
+        _connectors: &settings::Connectors,
+    ) -> CustomResult<Vec<(String, String)>, errors::ConnectorError> {
+        Ok(vec![])
+    }
+
+    fn get_content_type(&self) -> &'static str {
+        "application/json"
+    }
+
+    fn get_request_body(
+        &self,
+        req: &types::PaymentsCaptureRouterData,
+    ) -> CustomResult<Option<String>, errors::ConnectorError> {
+        let rapyd_req = utils::Encode::<rapyd::CaptureRequest>::convert_and_url_encode(req)
+            .change_context(errors::ConnectorError::RequestEncodingFailed)?;
+        Ok(Some(rapyd_req))
+    }
+
+    fn build_request(
+        &self,
+        req: &types::PaymentsCaptureRouterData,
+        connectors: &settings::Connectors,
+    ) -> CustomResult<Option<services::Request>, errors::ConnectorError> {
+        let timestamp = OffsetDateTime::unix_timestamp(OffsetDateTime::now_utc());
+        let salt = generate_id(12, "");
+
+        let rapyd_req = utils::Encode::<rapyd::CaptureRequest>::convert_and_encode(req)
+            .change_context(errors::ConnectorError::RequestEncodingFailed)?;
```

Comment:
You've called `convert_and_url_encode()` in `get_request_body()` and `convert_and_encode()` in `build_request()`. Is this intentional?

Reply:
missed it!!